ime/app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardBase.java

```diff
@@ -18,21 +18,33 @@
 
 import android.graphics.drawable.Drawable;
 import android.inputmethodservice.InputMethodService;
+import androidx.annotation.CallSuper;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.autofill.inline.UiVersions;
+import androidx.autofill.inline.v1.InlineSuggestionUi;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.util.Size;
+import android.util.TypedValue;
 import android.view.Gravity;
 import android.view.View;
 import android.view.ViewGroup;
 import android.view.Window;
 import android.view.WindowManager;
 import android.view.inputmethod.ExtractedText;
 import android.view.inputmethod.ExtractedTextRequest;
+import android.view.inputmethod.InlineSuggestion;
```

Comment:
The callbacks for inline suggestions have to be present in the InputMethodService. If you have any suggestions of a way forwards that removes this code from the base class, I'd be more than happy to try and implement your suggestion. I am very unfamiliar with the codebase so it may take me a lot of digging around to figure this out otherwise

Reply:
> This is amazing. Thank you for your contribution.
> 
> I have a few requests for change and I would also want you to add unit-tests for the new functionality.
> 
> Can you also attach some screenshots for the change?

screenshot has been uploaded to the OP

Reply:
So, the way I've structured the IME, was to encapsulate each of its distinct functionalities in a different base class. For example, everything related to dictionary suggestions is in [`AnySoftKeyboardSuggestions.`](https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/master/ime/app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardSuggestions.java). You'll notice that this class is extending `AnySoftKeyboardKeyboardSwitchedListener` which in turn extends another class all the way down to `AnySoftKeyboardBase` and all the way up to the concrete IME service `SoftKeyboard`.

My suggestion here was to create another class that goes somewhere in this hierarchy (let's say `AnySoftKeyboardInlineSuggestions` and have the related code in that new class. Makes sense?

Reply:
I can try and figure that out. It makes sense but I just need to figure out the structure of the app and how it all works. Is there any sort of documentation that I can read about the app's structure?

Reply:
I guess it's okay now. I'll probably move code around after the merge, if that's okay

Reply:
that's totally okay. Sorry, I know you're probably a busy guy, but I'm also just a bit lost in the structure lol. Super appreciate all the help you've offered as well as the work you've put into the app overall!

Reply:
I figured out the structure and put the AnySoftKeyboardInlineSuggestions after the AnySoftKeyboardSuggestions in the hierarchy. I appreciate the patience on this PR. I'll figure out the checks next

Reply:
Thank you for the effort. I'm excited for this change to land

Reply:
I think I got the tests passing. just running all the IME tests currently. I don't know if I hacked around the tests or if I am doing it correctly. I'll go ahead and run the full test suite after I confirm IME tests run and then I'll push those commits as long as the inline suggestions still show

Reply:
seeing some failed tests having to do with a `next_words_en.txt` file not being found. any ideas on this? I don't see that file anywhere in the project

Reply:
With my latest commits, I was able to get all unit tests passing, and the ci check script passing, and the app still shows the view as necessary! As long as everything looks good on your end, it looks like we can get this merged! Excited to see this come to fruition!